zcash_client_sqlite/src/wallet/commitment_tree.rs

```diff
@@ -1010,17 +1010,11 @@ pub(crate) fn put_shard_roots<
 
     let insert_into_cap = tracing::info_span!("insert_into_cap").entered();
     let cap_result = cap
-        .batch_insert(
+        .batch_insert::<(), _>(
             Position::from(start_index),
-            roots.iter().map(|r| {
-                (
-                    LevelShifter(r.root_hash().clone()),
-                    Retention::Checkpoint {
-                        id: (),
-                        is_marked: false,
-                    },
-                )
-            }),
+            roots
+                .iter()
+                .map(|r| (LevelShifter(r.root_hash().clone()), Retention::Reference)),
```

Comment:
Does the fact that we no longer checkpoint in the cap tree matter for backwards compatibility (serialization or otherwise)?

Reply:
It shouldn't. Checkpoint retention was possibly actually a bug, because it implied that pruning was permitted within the cap, which shouldn't be allowed to happen (it's not a correctness problem, because the cap nodes can always be recomputed, but losing that caching kind of defeats the purpose of the cap.)